# Calculating and then visualizing SNP density of the northern elephant seal genome by chromosome 
I will be running an analysis to calculate snp density of the northern elephant seal genome using a sliding window approach, then visualize in R by chromosome. I have adopted this method of visualizing whole-genome diversity from the SMSC class I attended last year :) yay! So that means this commande line + R code has been co-opted from code generated by Henrique Figuiero. 

I will be utilizing a combination of GATK and VCFTool commands to call variants then use a sliding window approach to calculate overall snp density. (will reword this later) 

# Use VCFTools to calculate snp density 

I've already produced a VCF file using GATK, so now I'm going to use VCFTools to call heterozygous sites.

    vcftools --gzvcf SRR25478315_bridgetrim.g.vcf.gz --recode --out SRR25478315_bridgetrim_only_hetsites.recode.vcf --maf 0.1

Use VCFTools to calculate snp density of the called heterozygous sites using a sliding window approach 

    vcftools --vcf SRR25478315_bridgetrim_only_hetsites.recode.vcf --SNPdensity 1000000 --out SRR25478315_bridgetrim_only_1mb_hetsites

# Visualize SNP Density by Chromosome in RStudio 


![Mirounga_heterozygous SNP densities 1Mb snpden](https://github.com/user-attachments/assets/c3871f9b-67af-45b7-ae2e-c34f4b4c1fbe)
